### ISSUE TITLE
Cryptic/inconsistent errors in most utilities

### ISSUE THREAD

**Issue by polymetric1** [redox-os/redox#905]
**Reproduction**: Pass any invalid arguments to any utility (`wget`, `cat`, 

**Expected behavior**: Utilities give useful and consistent errors such as "cat: testt: file doesn't exist"

**Actual behavior**: Utilities carrying out a similar task (such as finding and opening a file) seem to give very different errors from each other, and none of them are very clear as to what the error is (they all basically mean the file does not exist)

**Redox release**: 0.1.4

**Blocking/related**: None that I know of

**Misc**:
Some screenshots:
local files: (`tee` crashes the shell):
![virtualbox_2017-03-30_18-49-38](https://cloud.githubusercontent.com/assets/13613437/24529441/aa2e4bdc-1579-11e7-92f4-91d1a2563490.png)

When given an invalid URL, wget throws this crazy error:
![virtualbox_2017-03-30_18-42-55](https://cloud.githubusercontent.com/assets/13613437/24529445/b1a0e032-1579-11e7-9d76-f204bf66c88b.png)

**Comment by xTibor** [comment]
I don't think `tee` crashes the shell, it just waits for input from stdin. If you kill it from an another terminal you'll get your ion prompt back.

Some of the error messages can be improved by a small libextra tweaking (https://github.com/xTibor/libextra/commit/0b8e678a443aa2d727e82cca1446141fbceb3f39). However I'm not sure if that's the way to go.

**Comment by jackpot51** [comment]
Yes, fuck yes, libextra should be tweaked and **all** utilities should try to use it! My mistake has been putting unwraps() or custom error messages all over the place!

**Comment by polymetric1** [comment]
@xTibor True, I didn't think of that. But I'm pretty sure I tried typing and nothing happened. I'll check again.

**Comment by MggMuggins** [comment]
Seems like `unwrap_or_exit` is helpful for this, it allows short sourcecode and propagating specific errors up to the user. Error messages could theoretically improved by improving the quality of error messages coming out of the syscall crate.

Another thing that will help is clap-rs for utilities. For example, userutils is already ported, and that helps with validation of arguments. Might be good to define a set of options for clap-rs for redox utilities and get that all standard across the different repos.
